From c6abe5bc6310dbdf7684ef884f764049fd61c38d Mon Sep 17 00:00:00 2001
From: yingrenw <yingrenw@codeaurora.org>
Date: Thu, 6 Jul 2017 18:37:35 +0800
Subject: [PATCH 14/32] Launcher3: Can't search out local app by Chinese

Description:
Chinese content type is Character.OTHER_LETTER.
When key matches,it check the content type. And
if type is Character.OTHER_LETTER, it doesn't
support search and break.

When the content type is Character.OTHER_LETTER,
make it work normally.

Change-Id: I3713f24c9206fe16a8da2a23e6c90d68079dd533
Signed-off-by: Alexander Martinz <amartinz@shiftphones.com>
---
 .../launcher3/allapps/search/DefaultAppSearchAlgorithm.java    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/com/android/launcher3/allapps/search/DefaultAppSearchAlgorithm.java b/src/com/android/launcher3/allapps/search/DefaultAppSearchAlgorithm.java
index 26f6ec357..604ac8374 100644
--- a/src/com/android/launcher3/allapps/search/DefaultAppSearchAlgorithm.java
+++ b/src/com/android/launcher3/allapps/search/DefaultAppSearchAlgorithm.java
@@ -141,7 +141,8 @@ public class DefaultAppSearchAlgorithm implements SearchAlgorithm {
                 // Always a break point for a symbol
                 return true;
             default:
-                return  false;
+                // Always a break point at first character
+                return prevType == Character.UNASSIGNED;
         }
     }
 
-- 
2.17.1

